### ISSUE TITLE
Out of Memory Error - JVM heap size set to 8GB on OSX

### ISSUE THREAD

**Issue by hmgibson23** [ensime/ensime-server#520]
Possible duplicate of https://github.com/ensime/ensime-server/issues/293.

Running a particularly large project with ensime consistently runs out of memory, even with an enormous heap size of 8GB. 

Running Java 1.6.0 on OSX

Happens just after indexer, as follows:

```
[info] 11:27:58.572 [default-akka.actor.default-dispatcher-3] INFO  org.ensime.indexer.ClassFileIndex - Finished indexing 75 classpath files in 25.658 seconds.
[info] 11:27:58.573 [default-akka.actor.default-dispatcher-5] INFO  org.ensime.protocol.SwankProtocol - Writing: (:indexer-ready)
[error] java.lang.OutOfMemoryError: Java heap space
[error] java.lang.OutOfMemoryError: Java heap space

```
### Update

More specifically this issue happens once the index and analysis is complete and then I do anything in the file. Sometimes the error comes from a different Thread i.e Thread-2
